cvat/apps/authentication/auth.py

```diff
@@ -115,8 +115,8 @@ def is_job_owner(db_user, db_job):
 def is_job_annotator(db_user, db_job):
     db_task = db_job.segment.task
     # A job can be annotated by any user if the task's assignee is None.
-    has_rights = db_task.assignee is None or is_task_assignee(db_user, db_task)
-    if db_job.assignee is not None:
+    has_rights = (db_task.assignee is None and not settings.RESTRICTIONS['reduce_task_visibility']) or is_task_assignee(db_user, db_task)
+    if db_job.assignee is not None and not settings.RESTRICTIONS['reduce_task_visibility']:
```

Comment:
Why do we need to modify the line? If db_job.assignee isn't None probably the user should be able to view the task/job. Right?

Reply:
Agree.